x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationReloaderTests.java

```diff
@@ -632,6 +633,14 @@ private static CloseableHttpClient createHttpClient(SSLContext sslContext) {
             .build();
     }
 
+    private Settings.Builder getSettingsBuilder() {
```

Comment:
This approach is fragile as there is a high chance the next person that will write  a test that requires an SSLService will not be aware that not setting `DIAGNOSE_TRUST_EXCEPTIONS_SETTING` will cause FIPS 140 CI failures, but I didn't have any better ideas for this. An additional abstract class with the `getSettingsBuilder()` that extends `ESTestCase` so that all Test classes in x-pack plugin can extend that might be an alternative so that we don't need to define `getSettingsBuilder` all the time, but I'm not sure if this helps with being future-proof. `Settings.builder()` is too generic to add as a forbidden API.

Reply:
Also a hack, but you could extend `Settings.Builder` with a `put(...)` method that throws when passed `XPackSettings.DIAGNOSE_TRUST_EXCEPTIONS_SETTING.getKey()` and `true`?

Reply:
That wouldn't help much because true is the default value, so one needs not set it explicitly - thus that method wouldn't be called eitherway

Reply:
Given SunJsseFips is going away, I think we can live with this.
If it were going to stick around it would be worth finding some more reliable.